phoenix-core-client/src/main/java/org/apache/phoenix/compile/UpsertCompiler.java

```diff
@@ -842,11 +851,27 @@ public MutationPlan compile(UpsertStatement upsert) throws SQLException {
                         expression.getDataType(), column.getDataType(), "expression: "
                                 + expression.toString() + " in column " + column);
             }
+            if (!SchemaUtil.isPKColumn(column) && !isJsonNode(valueNode)) {
+                nonPKColumns.add(
+                        new Pair<>(ColumnName.caseSensitiveColumnName(column.getName().getString()),
```

Comment:
You are ignoring the column family name here. This is not correct

Reply:
Actually, can you verify this by adding a test case with the same column name in multiple column families.

Reply:
I see "SchemaUtil.isPKColumn(column)" being used at different places without checking to column family. What is the problem with multiple column families you find?

Reply:
The problem is not in `isPKColumn` but column.getName() which only returns the column qualifier

Reply:
Got it. Will add an IT to test these and add appropriate tests.